shell/platform/embedder/embedder.cc

```diff
@@ -712,6 +712,25 @@ inline flutter::PointerData::SignalKind ToPointerDataSignalKind(
   return flutter::PointerData::SignalKind::kNone;
 }
 
+// Returns the buttons for PointerData for the given buttons and change from a
+// FlutterPointerEvent.
+inline int64_t ToPointerDataButtons(
+    int64_t buttons,
+    flutter::PointerData::Change change) {
+  // `buttons` should be non-zero when and only when the pointer is down.
+  if (change == flutter::PointerData::Change::kDown ||
+      change == flutter::PointerData::Change::kMove) {
+    if (buttons == 0) {
+      // Synthesize primary button if the embedder fails to follow the contract.
+      // TODO(tongmu): Log a warning to inform the embedder to send the correct
```

Comment:
Could you change this from your username to a reference to the bug number for the bug where we were discussing this, since it provides more context on the circumstances where a warning should happen? As written here, without that context, it could encourage someone to add logging in cases where I don't think it's warranted (per my bug comments)